packages/buffetjs-core/src/components/DatePicker/index.js

```diff
@@ -35,134 +49,131 @@ class DatePicker extends React.PureComponent {
       displayedDate = date.format('MM/DD/YYYY');
     }
 
-    this.setState({ date, displayedDate });
-  }
-
-  // Temporary fix it the component will be migrated to react hooks in another PR
-  componentDidUpdate(prevProps) {
-    if (prevProps.value !== this.props.value) {
-      const { value, withDefaultValue } = this.props;
-      let date = null;
-      let displayedDate = '';
-
-      if (withDefaultValue && !value) {
-        date = moment();
-        displayedDate = date.format('MM/DD/YYYY');
-      }
-
-      if (!!value && moment(value).isValid()) {
-        date = value._isAMomentObject === true ? value : moment(value);
-        displayedDate = date.format('MM/DD/YYYY');
-      }
-
-      // Disabling this rule as it this component will be updated in the upcoming
-      // ui improvement PR
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ date, displayedDate });
+    dispatch({
+      type: 'SET_DATE',
+      date,
+    });
+
+    dispatch({
+      type: 'SET_DISPLAYED_DATE',
+      displayedDate,
+    });
+  }, []);
+
+  let timer = null;
+  const { date, displayedDate, isFocused, isVisible } = state;
+
+  const getDateValue = () => {
+    let dateValue = date ? date.format(displayFormat) : '';
+
+    if (isVisible) {
+      dateValue = displayedDate;
     }
-  }
 
-  handleChange = ({ target }) => {
-    this.setState({ displayedDate: target.value });
+    return dateValue;
+  };
+
+  const handleChange = ({ target }) => {
+    clearTimeout(timer);
+
+    dispatch({
+      type: 'SET_IS_FOCUSED',
+      isFocused: false,
+    });
 
-    clearTimeout(this.timer);
+    dispatch({
+      type: 'SET_DISPLAYED_DATE',
+      displayedDate: target.value,
+    });
 
-    this.timer = setTimeout(() => {
-      this.handleDateChange(moment(target.value));
-    }, this.props.wait);
+    timer = setTimeout(() => {
+      handleDateChange(moment(target.value, 'MM/DD/YYYY'));
+    }, wait);
   };
 
-  handleDateChange = date => {
-    const { name, onChange } = this.props;
+  const handleDateChange = dateValue => {
+    if (moment(dateValue).isValid()) {
+      onChange({ target: { name, type: 'date', value: dateValue } });
+
+      dispatch({
+        type: 'SET_DATE',
+        date: dateValue,
+      });
 
-    if (moment(date).isValid()) {
-      this.setState(
-        { date, displayedDate: date.format('MM/DD/YYYY'), visible: true },
-        () => onChange({ target: { name, type: 'date', value: date } })
-      );
+      dispatch({
+        type: 'SET_DISPLAYED_DATE',
+        displayedDate: dateValue.format('MM/DD/YYYY'),
+      });
     }
   };
 
-  handleFocusChange = ({ focused }) => {
-    /* istanbul ignore next */
-    this.setState({ isFocused: focused });
+  const handleDateClick = dateValue => {
+    handleDateChange(dateValue);
+
+    toggleDatepicker(false);
   };
 
-  handleOutsideClick = () => {
-    this.setState({
-      visible: false,
-    });
+  const handleOutsideClick = () => {
```

Comment:
I think this one is not really needed as you can directly use `toggleDatePicker(false)`